src/03-data-analysis-in-r/04-transform-1.Rmd

```diff
@@ -234,7 +234,55 @@ We've used the `mean()` and `sd()` functions above, but there are a host of othe
 
 ## The Pipe: Chaining Functions Together
 
-With the tools presented here we could do a decent job analyzing our `FAES` data. Let's say we wanted to subtract the mean of the `blank` from each `standard` signal and then summarize those results. It would look something like this: 
+Piping is a concept that allows you to chain functions together in a way that simplifies and clarifies your code. At its core, piping is similar to function composition in mathematics, where the output of one function becomes the input to the next. This helps you build complex operations in a readable and logical sequence.
+
+### Function Composition Example
+
+Consider a mathematical example of function composition:
+
+\[ f(g(x)) \]
+
+Here, the function \( g(x) \) is applied first, and its output is then passed as the input to the function \( f(x) \). In programming, this concept can be translated to chaining functions together.
+
+### Abstract Example of Piping
+
+The **pipe operator** `%>%`, an incredibly useful tool for writing more legible and understandable code. The pipe basically changes how you read code to emphasize the functions you're working with by passing the intermediate steps to hidden processes in the background.
+
+Now, consider the following abstract example of piping in R:
+
+```{r,eval=FALSE}
+result <- data %>%
+  step1() %>%
+  step2() %>%
+  step3()
+```
+
+Here’s what’s happening:
+
+- **`step1()`**: Takes `data` as its input.
+- **`step2()`**: Takes the result of `step1()` as its input.
+- **`step3()`**: Takes the result of `step2()` as its input.
+
+This sequence of operations could be written without pipes by nesting function calls, but the use of pipes makes the flow of data more explicit and easier to read.
+
+### A Simple Example of Piping
+
+Let’s start with a simple, single use of piping:
```

Comment:
The first example should show a function that takes no additional arguments, e.g. `FAES %>% nrow()`, and showing what it's equivalent to. Then you can keep what you currently have as a second example in this section. (So rename the section "Simple Examples of Piping".)